src/Apps/Artwork/Components/ArtworkImageBrowser/ArtworkActionsSaveButton.tsx

```diff
@@ -50,12 +52,32 @@ const ArtworkActionsSaveButton: React.FC<ArtworkActionsSaveButtonProps> = ({
     isLiveOpen ||
     registrationAttempted
 
+  const openAuctionRegistrationPopover = useCallback(() => {
+    if (!ignoreAuctionRegistrationPopover) {
+      setPopoverVisible(true)
+    }
+  }, [ignoreAuctionRegistrationPopover])
+
+  const onPopoverClosed = useCallback(() => {
+    setPopoverVisible(false)
+  }, [])
+
+  useEffect(() => {
+    mediator.on("artwork:save", openAuctionRegistrationPopover)
```

Comment:
Interesting that this is still useful — as we've been looking to remove this event bus for a while. Is there an easy way around not using it?

Reply:
Unfortunately, I didn't find another easy way. I tried this solution, but since the component is unmounted and `shouldDisplayPopover` is reset, it won't work 🤔

```javascript
const shouldDisplayPopover = useRef(false)

...

useEffect(() => {
  if (isLoggedIn && shouldDisplayPopover.current) {
    shouldDisplayPopover.current = false
    setPopoverVisible(true)
  }
}, [isLoggedIn])

...

<Popover {...other props} >
  {({ anchorRef }) => {
    return (
      <UtilButton
        {...other props}
        onClick={() => {
          handleSave()

          if (!ignoreAuctionRegistrationPopover) {
            if (isLoggedIn) {
              setPopoverVisible(true)
              return
            }

            shouldDisplayPopover.current = true
          }
        }}
      />
    )
  }}
</Popover>
```

Reply:
Yeah, could do something with a query param perhaps. But I guess I'm not opposed to us having an event bus — it's a little weird though in terms of patterns that exist. The more React way would be to set something in the system context or some other parent context.

Reply:
If you don't mind, then I'll leave the current solution and on one of the feature Friday I'll see what can be done with an event bus

Reply:
Sounds good to me! Thanks!